parser/xml/src/test/java/org/codice/ddf/parser/xml/TestXmlParser.java

```diff
@@ -0,0 +1,228 @@
+        os = new ByteArrayOutputStream();
+        parser.marshal(configurator, mother, os);
+        outputXml = os.toString();
+
+        assertXpathEvaluatesTo("Padme", "/mother/@firstname", outputXml);
+    }
+
+    @Test
+    public void testUnmarshal() throws Exception {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        parser.marshal(configurator, mother, os);
+
+        ByteArrayInputStream is = new ByteArrayInputStream(os.toByteArray());
+        MotherElement unmarshal = parser.unmarshal(configurator, MotherElement.class, is);
+
+        assertEquals(mother.getAge(), unmarshal.getAge());
+        assertEquals(mother.getFirstname(), unmarshal.getFirstname());
+        assertEquals(mother.getLastname(), unmarshal.getLastname());
+        assertEquals(mother.getChild().size(), unmarshal.getChild().size());
+        assertEquals(luke.getFirstname(), unmarshal.getChild().get(0).getFirstname());
+        assertEquals(leia.getAge(), unmarshal.getChild().get(1).getAge());
+
+        configurator.setHandler(new DefaultValidationEventHandler());
+        is = new ByteArrayInputStream(os.toByteArray());
+        unmarshal = parser.unmarshal(configurator, MotherElement.class, is);
+
+        assertEquals(mother.getAge(), unmarshal.getAge());
+
+        configurator.addProperty("UnknownProperty", Boolean.TRUE);
+        is = new ByteArrayInputStream(os.toByteArray());
+        thrown.expect(ParserException.class);
+        unmarshal = parser.unmarshal(configurator, MotherElement.class, is);
+    }
+
+    @Test
+    public void testUnmarshalBadCast() throws Exception {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        parser.marshal(configurator, mother, os);
+
+        ByteArrayInputStream is = new ByteArrayInputStream(os.toByteArray());
+
+        thrown.expect(ClassCastException.class);
+        ChildElement unmarshal = parser.unmarshal(configurator, ChildElement.class, is);
+    }
+
+    @Test
+    public void testBadContextPath() throws Exception {
+        configurator.setContextPath(ImmutableList.of(""));
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+        thrown.expect(ParserException.class);
+        parser.marshal(configurator, mother, os);
+    }
+
+    @Test
+    public void testBadMarshal() throws Exception {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+        thrown.expect(ParserException.class);
+        parser.marshal(configurator, this, os);
+    }
+
+    @Test
+    public void testBadUnmarshal() throws Exception {
+        ByteArrayInputStream is = new ByteArrayInputStream(new byte[] {0, 1, 2});
+
+        thrown.expect(ParserException.class);
+        ChildElement unmarshal = parser.unmarshal(configurator, ChildElement.class, is);
+    }
+
+    @Test
+    public void testTypeAdapter() throws Exception {
+        // TODO RAP 30 Jun 15: Actually need to *test* the type adapter
```

Comment:
I certainly could but my thinking was that if we wanted to go forward with this, I'd actually write that test correctly and not ignore it at all.